test/AWS.Deploy.CLI.IntegrationTests/ServerModeTests.cs

```diff
@@ -0,0 +1,227 @@
+            {
+                var baseUrl = $"http://localhost:{portNumber}/";
+                var restClient = new RestAPIClient(baseUrl, httpClient);
+                
+                await WaitTillServerModeReady(restClient);
+
+                var startSessionOutput = await restClient.StartDeploymentSessionAsync(new StartDeploymentSessionInput
+                {
+                    AwsRegion = _awsRegion,
+                    ProjectPath = projectPath
+                });
+
+                var sessionId = startSessionOutput.SessionId;
+                Assert.NotNull(sessionId);
+
+                var signalRClient = new DeploymentCommunicationClient(baseUrl);
+                await signalRClient.JoinSession(sessionId);
+
+                var logOutput = new StringBuilder();
+                signalRClient.ReceiveLogAllLogAction = (line) =>
+                {
+                    logOutput.AppendLine(line);
+                };
+
+                var getRecommendationOutput = await restClient.GetRecommendationsAsync(sessionId);
+                Assert.NotEmpty(getRecommendationOutput.Recommendations);
+
+                var fargateRecommendation = getRecommendationOutput.Recommendations.FirstOrDefault(x => string.Equals(x.RecipeId, "AspNetAppEcsFargate"));
+                Assert.NotNull(fargateRecommendation);
+
+                await restClient.SetDeploymentTargetAsync(sessionId, new SetDeploymentTargetInput
+                {
+                    NewDeploymentName = _stackName,
+                    NewDeploymentRecipeId = fargateRecommendation.RecipeId
+                });
+
+                await restClient.StartDeploymentAsync(sessionId);
+
+                await WaitForDeployment(restClient, sessionId);
+
+                var stackStatus = await _cloudFormationHelper.GetStackStatus(_stackName);
+                Assert.Equal(StackStatus.CREATE_COMPLETE, stackStatus);
+
+                Assert.True(logOutput.Length > 0);
+                Assert.Contains("Initiating deployment", logOutput.ToString());
+            }
+            finally
+            {
+                cancelSource.Cancel();
+                await _cloudFormationHelper.DeleteStack(_stackName);
+                _stackName = null;
+            }
+        }
+
+        private async Task<DeploymentStatus> WaitForDeployment(RestAPIClient restApiClient, string sessionId)
+        {
+            DeploymentStatus status = DeploymentStatus.NotStarted;
+            do
+            {
+                try
+                {
+                    await Task.Delay(2000);
+                    status = (await restApiClient.GetDeploymentStatusAsync(sessionId)).Status;
+                }
+                catch (Exception)
+                {
+
+                }
+            } while (status == DeploymentStatus.Executing);
+
+            return status;
+        }
+
+
+        private async Task WaitTillServerModeReady(RestAPIClient restApiClient)
+        {
+            int attempt = 0;
```

Comment:
attempt is never incremented.

Reply:
We should use something generic for retries like [this](https://github.com/aws/aws-dotnet-deploy/blob/main/test/AWS.Deploy.CLI.IntegrationTests/Helpers/WaitUntilHelper.cs#L18).